File: androidApp/src/main/java/com/mecofarid/trending/ui/repo/TrendingViewModel.kt
package com.mecofarid.trending.ui.repo

import androidx.lifecycle.asLiveData
import com.mecofarid.trending.domain.features.repo.domain.interactor.GetTrendingInteractor

class TrendingViewModel(
    repoInteractor: GetTrendingInteractor
): CommonTrendingViewModel(repoInteractor){
    val liveUiState = uiState.asLiveData()
}

File: iosApp/iosApp/UI/Repo/TrendingViewModel.swift
//
//  TrendingViewModel.swift
//  Trending
//
//  Created by Farid Mammadov on 05.01.23.
//

import Foundation
import shared

@MainActor
class TrendingViewModel: CommonTrendingViewModel, ObservableObject {

    @Published private(set) var liveUiState: State!

    override init(repoInteractor: GetTrendingInteractor) {
        super.init(repoInteractor: repoInteractor)
        observeUiState{ state in
            self.liveUiState = state
        }
    }

    deinit {
        clear()
    }
}
